### DISCUSSION TITLE
Run the project without uploading node_modules

### DISCUSSION THREAD

**Discussion by goc-chushixian** [vercel/next.js#39600] [Help] [answered]
After I build my project, I need to upload my node_modules together so that the project can run. Is there any way not to upload node_modules

**Comment by gaston-flores** [comment, marked as answer]
You can use [`output: 'standalone'`](https://nextjs.org/docs/advanced-features/output-file-tracing#automatically-copying-traced-files) in your `next.config.js` to generate a build that can be deployed on its own without installing `node_modules`. Check out the [with-docker example](https://github.com/vercel/next.js/tree/canary/examples/with-docker).

**Reply by goc-chushixian** [reply]
Thank you. That would solve my problem

**Reply by Mohammad961harb** [reply]
Hello sir , can i also use this feature even if I use a custom server ?

**Reply by goc-chushixian** [reply]
This generates a server.js. I don't think it's a problem for you to revise on this basis
